### PR TITLE
Move d_do_test.d combination into a function.

### PR DESCRIPTION
**Pull request by marler8997**
This is in preparation for #7099.  By moving the code for a single test combination into a function, the logic that drives all the combinations can be enhanced without having to increase the nesting of the combination code itself.

### CODE REVIEW

**Review by wilzbach — APPROVED** [review]
Looks great. Thanks!

File: test/d_do_test.d [hunk 1]
```diff
@@ -668,7 +669,7 @@ int tryMain(string[] args)
                 }
                 else version (linux)
                 {
-                    auto script = test_app_dmd_base ~ to!string(i) ~ ".gdb";
+                    auto script = test_app_dmd_base ~ to!string(permuteIndex) ~ ".gdb";
```
Comment (wilzbach) [inline test/d_do_test.d]: You need to update the `version(Windows)` block above too.

**Review by wilzbach — DISMISSED** [review]
Sorry - that was a bit hard to catch.

File: test/d_do_test.d (hunk 1 above)
Comment (marler8997) [reply test/d_do_test.d]: done

**Review by wilzbach — APPROVED** [review]
I will wait for another pairs of eyes then.

**Comment by stefan-koch-sociomantic** [comment]
that code is rather unclear due to the lack of documentation. 
If a function is doing something that is not trivial please share it in a sentence or two.

**Comment by marler8997** [comment]
Ok I added a comment for the `testCombination` function.  Since this PR is fairly trivial and blocking #7099, could we get this merged sooner rather than later?

**Pull request merged**